### ISSUE TITLE
Connection from second instance to S7 could not be established

### ISSUE THREAD

**Issue by Useserall** [iobroker-community-adapters/ioBroker.mihome-vacuum#499]
**Describe the bug**  
Hello, at first thanks for your efforts to create this adapter and keep it up to date! 😃 

The problem I currently have is that it is not possible to connect to my second vacuum cleaner (a15 / S7). I tried to add a second cleaner in a second instance. To do that I added the cleaner in the Mi Home app. In the second instance I added my credentials to retreive the new cleaner with the corresponding token and changed the local port. Everything worked fine until here. I added the cleaner with the token and local IP adress. The problem is that the adapter could not connect to the cleaner. The debug message is all the time:
"Helo message Timeout"

The iobroker instance and cleaner are in different subnets. There are also firewall rules in place, but the iobroker should have access to both vacuum cleaners. My old cleaner (v1) is also connected and works fine. Also no dropped traffic is shown in the firewall log file. I tried also to disable the corresponding firewall rule.

I don't know what I could try further to solve the issue...
**To Reproduce**  

1. Add the cleaner IP, token etc. in the adapter
2. Instance is not able to connect to the cleaner

**Expected behavior**  
Instance should be connected to the cleaner (like for my first vacuum cleaner)

**Screenshots & Logfiles**  
<html>
<body>

Adapter | Time | Log | Message
-- | -- | -- | --
mihome-vacuum.1 | 2022-01-30 14:32:44.641 | debug | Helo message Timeout
mihome-vacuum.1 | 2022-01-30 14:32:32.637 | debug | Helo message Timeout
mihome-vacuum.1 | 2022-01-30 14:32:20.633 | debug | Helo message Timeout
mihome-vacuum.1 | 2022-01-30 14:32:08.617 | debug | Helo message Timeout
mihome-vacuum.1 | 2022-01-30 14:32:06.611 | debug | server started on 0.0.0.0:53422
mihome-vacuum.1 | 2022-01-30 14:32:06.604 | info | IOT enabled, create state
mihome-vacuum.1 | 2022-01-30 14:32:06.593 | debug | MIIO: Config: ip:10.0.200.17 token: xxxxxxxxxxxxxxxxxx
mihome-vacuum.1 | 2022-01-30 14:32:06.590 | debug | Create State for deviceInfowifi_signal
mihome-vacuum.1 | 2022-01-30 14:32:06.589 | debug | Create State for deviceInfofw_ver
mihome-vacuum.1 | 2022-01-30 14:32:06.587 | debug | Create State for deviceInfomodel
mihome-vacuum.1 | 2022-01-30 14:32:06.582 | debug | Create State for deviceInfomac
mihome-vacuum.1 | 2022-01-30 14:32:06.580 | debug | Create State for deviceInfo
mihome-vacuum.1 | 2022-01-30 14:32:06.399 | info | starting. Version 3.3.3 in /opt/iobroker/node_modules/iobroker.mihome-vacuum, node: v14.18.3, js-controller: 3.3.22

</body>
</html>

</body>
</html>

**Versions:**

-   Adapter version: 3.3.3
-   JS-Controller version:  3.3.22
-   Node version: v14.18.3
-   vacuum cleaner: a15 / S7

**Comment by Useserall** [comment]
What I forgot to say is, that I am also able to ping the cleaner from the iobroker instance/server.

**Comment by thedd1337** [comment]
I have the same issue but it is not S7 related
I change my first instance to use my second robot S7 and wanted to use the second instance for my S5

The S5 was running fine in first instance, in second i always see "Helo message Timeout"
The S7 was not running in second instance but as i choose it in the first one it is working fine now.

So i think there might be a general issue with the ports used by the second instance

**Comment by Useserall** [comment]
Hello,

I updated the adapter to version v3.4.0.
The problem still exits, nothing changed in the past time.


Regards!

**Comment by dirkhe** [comment]
Are the s7 and ioBroker in the same subnet? If no, there is a problem with the s7. The adapter has no influence here. See #577

**Comment by thedd1337** [comment]
I can proove issue still persists even if iobroker and both robots are in same subnet as in my environment

**Comment by Useserall** [comment]
@dirkhe, so in my case the s7 and ioBroker are not in the same subnet. I also saw the mentioned issue a few minutes afterwards. Thanks that you shared the information here!

And @thedd1337 interesting.. So, you still have the problem although both are in the same subnet? You tried to use the S7 in the working instance?

**Comment by dirkhe** [comment]
I guess the adapter can't solve the problem, because it is a timeout, meaning your robot doesn't answer So please try 
`nc -z -v -u [hostname/IP address] [port number]`

**Comment by thedd1337** [comment]
> I guess the adapter can't solve the problem, because it is a timeout, meaning your robot doesn't answer So please try `nc -z -v -u [hostname/IP address] [port number]`

Well i would say the description for a second robot in the adapter was my problem as it is not total clear what port has to be changed

I tested both robots and found the issue.
It was the Port of the robot, as i changed it,too. 
I changed it back to default and only changed to port of the instance to another one.

In the documentation i read:
_Zweiter Roboter
Sollen zwei Roboter über ioBroker gesteuert werden, müssen zwei Instanzen angelegt werden. Der zweite Roboter muss seinen eigenen Port (Standard: 53421) ändern, damit beide Roboter unterschiedliche Ports haben._

There should be written: Bei einem zweiten Roboter muss der eigene Port geändert werden, der des Roboters muss  54321 bleiben
